compiler/src/iree/compiler/Codegen/Common/IREEComprehensiveBufferizePass.cpp

```diff
@@ -165,10 +165,38 @@ eliminateEmptyTensors(RewriterBase &rewriter, Operation *op,
   return success();
 }
 
+/// Multiple uses of `tensor.empty()` results in a copy since upstream
+/// treats `tensor.empty()` as an allocation and sees uses as a data-hazard
+/// creating copies/allocations. Since the `empty` op is a proxy for
+/// undef, these could just be duplicated to have a single use. This removes
+/// unnecessary data-hazards.
+static LogicalResult duplicateTensorEmptyOps(OpBuilder &b,
```

Comment:
Can you refactor it to `Utils/Utils.[h|cpp]`?

https://github.com/iree-org/iree/blob/dd3f2a392819d121fa5329a1c591be06ae9e887a/compiler/src/iree/compiler/Codegen/Utils/Utils.h#L208-L211

Reply:
Please treat it as a draft patch for now. I am just testing things on CI.